src/store/inflight_activation.rs

```diff
@@ -290,11 +444,85 @@ impl InflightActivationStore {
         Ok(Some(row.into()))
     }
 
+    pub async fn get_activation_raw_blob(&self, id: &str) -> Result<Vec<u8>, Error> {
+        if self.is_new_read() {
+            let blob_result: Option<BlobTableRow> = sqlx::query_as(
+                "
+                SELECT id, activation, added_at
+                FROM taskactivation_blobs 
+                WHERE id = $1
+                ",
+            )
+            .bind(id)
+            .fetch_optional(&self.blob_read_pool)
+            .await?;
+
+            match blob_result {
+                Some(blob_row) => Ok(blob_row.activation),
+                None => Err(anyhow!("could not find blob for activation")),
+            }
+        } else {
+            let inflight = self.get_meta_by_id(id).await?.unwrap();
+            Ok(inflight.activation.unwrap().encode_to_vec())
+        }
+    }
+
+    pub async fn get_activation_blob(&self, id: &str) -> Result<TaskActivation, Error> {
+        let activation = self.get_activation_raw_blob(id).await?;
+        let activation_data: &[u8] = &activation;
+        Ok(TaskActivation::decode(activation_data)?)
+    }
+
     #[instrument(skip_all)]
     pub async fn store(&self, batch: Vec<InflightActivation>) -> Result<QueryResult, Error> {
         if batch.is_empty() {
             return Ok(QueryResult { rows_affected: 0 });
         }
+
+        // A note about the consistency of the data in the two databases:
+        // If there is data in the blob database that isn't in the meta database, that is not
+        // a consistency issue. No extra tasks will get run or failed. The orphaned blobs
+        // will get cleaned up. However if there is data in the meta database that is not reflected
+        // in the blob database, that is a consistency error. As such, write the blobs first and
+        // that succeeds, then attempt to write the meta. A failure on either write will cause the
+        // entire batch to be failed.
+        if self.is_new_write() {
+            let mut blob_query_builder = QueryBuilder::<Sqlite>::new(
+                "
+                INSERT INTO taskactivation_blobs
+                    (
+                        id,
+                        activation,
+                        added_at
+                    )
+                ",
+            );
+            let blob_rows = batch
+                .clone()
+                .into_iter()
+                .map(BlobTableRow::try_from)
+                .collect::<Result<Vec<BlobTableRow>, _>>()?;
+            let blob_query = blob_query_builder
+                .push_values(blob_rows, |mut b, row| {
+                    b.push_bind(row.id);
+                    b.push_bind(row.activation);
+                    b.push_bind(row.added_at.timestamp());
+                })
+                .push(" ON CONFLICT(id) DO NOTHING")
```

Comment:
Just a note on `ON CONFLICT(id) DO NOTHING` since I think it has an interesting implication down the road. We know that it's possible for writes to the blob db to succeed and writes to meta db to fail. If a batch fails and is retried later, the blob will not be updated since the ID already exists. This seems ok since the blob doesn't store any information that gets altered by `inflight_activation_writer` when it gets retried.